tracer/src/Datadog.AutoInstrumentation.NativeLoader/instrumented_assembly_generator/instrumented_assembly_generator_metadata_interfaces.cpp

```diff
@@ -0,0 +1,1734 @@
+#include "instrumented_assembly_generator_metadata_interfaces.h"
+#include "../log.h"
+#include "instrumented_assembly_generator_helper.h"
+
+namespace instrumented_assembly_generator
+{
+    //TODO:
+// The behaviour of the IMetadata interface when defining members is to creat ea new one if no one exist, and return the
+// existing one if it's already exist
+// so by definition we may write to disk more than one identical member. To avoid this, we have to keep hashset of the
+// created tokens.
+
+InstrumentedAssemblyGeneratorMetadataInterfaces::InstrumentedAssemblyGeneratorMetadataInterfaces(
+    const ComPtr<IUnknown>& metadataInterfaces, ICorProfilerInfo12* corProfilerInfo) :
+    m_metadataInterfaces(metadataInterfaces), m_corProfilerInfo(corProfilerInfo)
+{
+    Log::Debug("InstrumentedAssemblyGeneratorMetadataInterfaces::.ctor");
+    // attach from tests, we can do it also with ev and spin
+    //std::this_thread::sleep_for(std::chrono::milliseconds(20000));
+
+    AddRef();
+}
+
+InstrumentedAssemblyGeneratorMetadataInterfaces::~InstrumentedAssemblyGeneratorMetadataInterfaces()
+{
```

Comment:
no release ?

Reply:
The usage of this class is through `ComPtr` so the `Release()` will be called on the caller's side (for example [here](https://github.com/DataDog/dd-trace-dotnet/pull/2829/files#diff-11af2f425a820638208ab0234332c17d618a959c096c6ccf9e2572b93f28d211R73)), also the [member of the class](https://github.com/DataDog/dd-trace-dotnet/pull/2829/files#diff-48d1435cc2f813026c99c289c44180bfd99599b1b9f138e4997d11f78dc6f7acR23) is defined as a `ComPtr` and so its destructor (which calls `Release()`) will be called automatically so as I understand no need to call release here. I tried to add a release call but it cause the process to crash, which seems to be good indication that this would be one decrement too much. But maybe I'm wrong, WDYT?